### PR TITLE
EIRH-96: Mixed conjunctive and disjunctive filters.

### PR DESCRIPTION
**Pull request by sherakama**
# INITIAL REVIEW

# Summary
- Changes the logic around so that venues and vendors are in conjunctive groups
- Changes the logic around so that policies is all disjunctive
- Handles the attribute refinement in the customRefinement connector
- Related documentation: https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/

# Review By (Date)
- EOW

# Criticality
- High

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch on https://eirhtable.sites.stanford.edu/discovery ✅ (done)
2. Load up the UI at https://eirhtable.sites.stanford.edu/discovery
3. Test the filters on each of the three tabs to see if they are returning the results you'd expect
4. Verify the state of the filters carries over from one tab to the next and the results in each tab are correct
5. Verify the deep link works

## TODO

- [ ] Add language to page about how to use the filters on each tab
- [ ] Test combinations of filters and other stuff
- [ ] Clean up codes
- [ ] Validate the Any/All special case is working appropriately

### CODE REVIEW

**Comment by sherakama** [comment]
TLDR; If were going to switch to conjunctive facets with the groupings the data shape will look like

```
[ 'single', ['A', B', 'C'], ['C', 'F', X'], 'A', ['G', 'C'] ]
```

Where this translates into search results where...

The result is 'single' AND (A,B, OR C) AND (C, F, OR X) AND 'A' AND (G OR C)

**Review by rebeccahongsf — APPROVED** [review]
Works as expected in the preview 👍
